### PR TITLE
Support bitrate, user limit, and permission overwrites in creating channels

### PR DESCRIPTION
**Pull request by z64**
This PR primarily replaces the OpenStruct that was used to represent channel permission overwrites and replaces it with an actual data object.

This lets you ship channels easily with overwrites already on them, for example, to stop @SunDwarf 's bot from being able to post "first" in the channel before you can apply an overwrite to block it.

In addition, when creating voice channels, you can instance them with `bitrate` and `user limit` already set instead of having to make a subsequent call.

#309

### CODE REVIEW

File: lib/discordrb/data.rb [hunk 1]
```diff
@@ -1154,6 +1215,32 @@ def group?
...
+    def permission_overwrites(type = nil)
+      return @permission_overwrites unless type
+      @permission_overwrites.values.select { |e| e.type == type }
+    end
+
+    alias_method :overwrites, :permission_overwrites
+
+    # @return [Overwrite] any member-type permission overwrites on this channel
+    def member_overwrites
+      permission_overwrites 'member'
```
Comment (meew0) [inline lib/discordrb/data.rb]: Nitpick: a symbol should be used here instead of creating a string every time

File: lib/discordrb/data.rb [hunk 2]
```diff
@@ -1032,6 +1031,78 @@ def url
...
+    #   Overwrite.new(120571255635181568, type: 'member', allow: 1024, deny: 0)
+    # @param object [Integer, #id] the ID or object this overwrite is for
+    # @param type [String] the type of object this overwrite is for (only required if object is an Integer)
+    # @param allow [Integer, Permissions] allowed permissions for this overwrite, by bits or a Permissions object
+    # @param deny [Integer, Permissions] denied permissions for this overwrite, by bits or a Permissions object
+    def initialize(object = nil, type: nil, allow: 0, deny: 0)
+      @id = object.respond_to?(:id) ? object.id : object
+
+      @type = if object.is_a? Member
+                'member'
```
Comment (meew0) [inline lib/discordrb/data.rb]: Same here (symbols instead of strings)

**Comment by SunDwarf** [comment]
Don't oppress my use cases

**Comment by meew0** [comment]
There appears to be one more problem: Discord's data has the `type` as a string, but your code (data.rb:1070) checks for whether it's a symbol. This makes the initialisation fail.

**Comment by z64** [comment]
Initialization tested fixed now. Will take whatever can `#to_sym`

**Comment by meew0** [comment]
Whoops, totally forgot to merge this after resolving the conflicts, sorry

**Pull request merged**